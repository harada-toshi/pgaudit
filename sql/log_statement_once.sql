\! cp sql/log_statement_once/log_statement_once_default.conf /tmp/config.conf
\! echo "pgaudit.config_file = '/tmp/config.conf'" >> ./tmp_check/data/postgresql.conf
\! rm ./tmp_check/tmp_check.log

\! pg_ctl restart -w -D ./tmp_check/data -l ./tmp_check/tmp_check.log
\! psql contrib_regression -q -a < sql/log_statement_once/log_statement_once.sql
\! rm /tmp/config.conf

\! cp sql/log_statement_once/log_statement_once_on.conf /tmp/config.conf
\! pg_ctl restart -w -D ./tmp_check/data -l ./tmp_check/tmp_check.log
\! psql contrib_regression -q -a < sql/log_statement_once/log_statement_once.sql
\! rm /tmp/config.conf

\! cp sql/log_statement_once/log_statement_once_off.conf /tmp/config.conf
\! pg_ctl restart -w -D ./tmp_check/data -l ./tmp_check/tmp_check.log
\! psql contrib_regression -q -a < sql/log_statement_once/log_statement_once.sql
\! rm /tmp/config.conf

\! grep "OBJECT" ./tmp_check/tmp_check.log > results/log_statement_once.out

