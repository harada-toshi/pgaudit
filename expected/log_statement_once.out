LOG:  AUDIT: OBJECT,8,1,READ,SELECT,TABLE,public.foo,SELECT * FROM foo;,<not logged>
LOG:  AUDIT: OBJECT,9,1,READ,SELECT,TABLE,public.foo,"SELECT foo.id, bar.data FROM foo, bar WHERE foo.id = bar.id;",<not logged>
LOG:  AUDIT: OBJECT,9,1,READ,SELECT,TABLE,public.bar,"SELECT foo.id, bar.data FROM foo, bar WHERE foo.id = bar.id;",<not logged>
LOG:  AUDIT: OBJECT,10,1,READ,SELECT,TABLE,public.foo,"SELECT foo.id, bar.data, baz.data FROM foo, bar, baz WHERE foo.id = bar.id AND foo.id = baz.id;",<not logged>
LOG:  AUDIT: OBJECT,10,1,READ,SELECT,TABLE,public.bar,"SELECT foo.id, bar.data, baz.data FROM foo, bar, baz WHERE foo.id = bar.id AND foo.id = baz.id;",<not logged>
LOG:  AUDIT: OBJECT,10,1,READ,SELECT,TABLE,public.baz,"SELECT foo.id, bar.data, baz.data FROM foo, bar, baz WHERE foo.id = bar.id AND foo.id = baz.id;",<not logged>
LOG:  AUDIT: OBJECT,8,1,READ,SELECT,TABLE,public.foo,SELECT * FROM foo;,<not logged>
LOG:  AUDIT: OBJECT,9,1,READ,SELECT,TABLE,public.foo,"SELECT foo.id, bar.data FROM foo, bar WHERE foo.id = bar.id;",<not logged>
LOG:  AUDIT: OBJECT,9,1,READ,SELECT,TABLE,public.bar,<previously logged>,<previously logged>
LOG:  AUDIT: OBJECT,10,1,READ,SELECT,TABLE,public.foo,"SELECT foo.id, bar.data, baz.data FROM foo, bar, baz WHERE foo.id = bar.id AND foo.id = baz.id;",<not logged>
LOG:  AUDIT: OBJECT,10,1,READ,SELECT,TABLE,public.bar,<previously logged>,<previously logged>
LOG:  AUDIT: OBJECT,10,1,READ,SELECT,TABLE,public.baz,<previously logged>,<previously logged>
LOG:  AUDIT: OBJECT,8,1,READ,SELECT,TABLE,public.foo,SELECT * FROM foo;,<not logged>
LOG:  AUDIT: OBJECT,9,1,READ,SELECT,TABLE,public.foo,"SELECT foo.id, bar.data FROM foo, bar WHERE foo.id = bar.id;",<not logged>
LOG:  AUDIT: OBJECT,9,1,READ,SELECT,TABLE,public.bar,"SELECT foo.id, bar.data FROM foo, bar WHERE foo.id = bar.id;",<not logged>
LOG:  AUDIT: OBJECT,10,1,READ,SELECT,TABLE,public.foo,"SELECT foo.id, bar.data, baz.data FROM foo, bar, baz WHERE foo.id = bar.id AND foo.id = baz.id;",<not logged>
LOG:  AUDIT: OBJECT,10,1,READ,SELECT,TABLE,public.bar,"SELECT foo.id, bar.data, baz.data FROM foo, bar, baz WHERE foo.id = bar.id AND foo.id = baz.id;",<not logged>
LOG:  AUDIT: OBJECT,10,1,READ,SELECT,TABLE,public.baz,"SELECT foo.id, bar.data, baz.data FROM foo, bar, baz WHERE foo.id = bar.id AND foo.id = baz.id;",<not logged>
